zookeeper-server/src/main/java/org/apache/zookeeper/cli/DelQuotaCommand.java

```diff
@@ -129,7 +137,23 @@ public static boolean delQuota(ZooKeeper zk, String path,
             }
             // cut the tree till their is more than one child
             trimProcQuotas(zk, parentPath);
+        } else {
+            if (quota.getCount() > 0) {
+                strack.setCount(-1);
+            }
+            if (quota.getBytes() > 0) {
+                strack.setBytes(-1L);
+            }
+            if (quota.getCountHardLimit() > 0) {
+                strack.setCountHardLimit(-1);
+            }
+            if (quota.getByteHardLimit() > 0) {
+                strack.setByteHardLimit(-1L);
+            }
+
+            zk.setData(quotaPath, strack.toString().getBytes(), -1);
```

Comment:
Same US-ASCII remark as above, but for [`getBytes`](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#getBytes-java.nio.charset.Charset-):

```java
strack.toString().getBytes(US_ASCII)
```

with, perhaps, a new `byte[] StatsTrack.toNodeData()` method which centralizes the conversion in one place?

Reply:
Add a new method `public byte[] getStatsBytes()`